dialect/sql/sqlgraph/graph_test.go

```diff
@@ -1955,6 +1956,73 @@ func TestQueryEdgesSchema(t *testing.T) {
 	require.Equal(t, [][]int64{{4, 5}, {4, 6}}, edges)
 }
 
+func TestIsConstraintError(t *testing.T) {
+	testCases := []struct {
+		name               string
+		errMessage         string
+		expectedConstraint bool
+		expectedFK         bool
+		expectedUnique     bool
+	}{
+		{
+			name: "MySQL FK",
+			errMessage: "insert node to table \"pets\": Error 1452: Cannot add or update a child row: a foreign key" +
+				" constraint fails (`test`.`pets`, CONSTRAINT `pets_users_pets` FOREIGN KEY (`user_pets`) REFERENCES " +
+				"`users` (`id`) ON DELETE SET NULL)",
+			expectedConstraint: true,
+			expectedFK:         true,
+			expectedUnique:     false,
+		},
+		{
+			name:               "SQLite FK",
+			errMessage:         "insert node to table \"pets\": FOREIGN KEY constraint failed",
+			expectedConstraint: true,
+			expectedFK:         true,
+			expectedUnique:     false,
+		},
+		{
+			name:               "Postgres FK",
+			errMessage:         "insert node to table \"pets\": pq: insert or update on table \"pets\" violates foreign key constraint \"pets_users_pets\"",
+			expectedConstraint: true,
+			expectedFK:         true,
+			expectedUnique:     false,
+		},
+		{
+			name: "MySQL Unique",
+			errMessage: "insert node to table \"file_types\": UNIQUE constraint failed: file_types.name ent: constraint" +
+				" failed: insert node to table \"file_types\": UNIQUE constraint failed: file_types.name",
+			expectedConstraint: true,
+			expectedFK:         false,
+			expectedUnique:     true,
+		},
+		{
+			name: "SQLite Unique",
+			errMessage: "insert node to table \"file_types\": UNIQUE constraint failed: file_types.name ent: " +
+				"constraint failed: insert node to table \"file_types\": UNIQUE constraint failed: file_types.name",
+			expectedConstraint: true,
+			expectedFK:         false,
+			expectedUnique:     true,
+		},
+		{
+			name: "Postgres Unique",
+			errMessage: "insert node to table \"file_types\": pq: duplicate key value violates unique constraint" +
+				" \"file_types_name_key\" ent: constraint failed: insert node to table \"file_types\": " +
+				"pq: duplicate key value violates unique constraint \"file_types_name_key\"",
+			expectedConstraint: true,
+			expectedFK:         false,
+			expectedUnique:     true,
+		},
+	}
+	for _, testCase := range testCases {
```

Comment:
Keep the same style we have in this file.
```suggestion
	for _, tt := range tests {
```